### DISCUSSION TITLE
Samsung Watch 5 (Wear OS)

### DISCUSSION THREAD

**Discussion by franzbischoff** [ESPresense/ESPresense#718] [General]
Just to inform. It works!

1. The IRK is broadcasted when you turn on the BT on the device
2. BT MAC seems fixed when you use Android App scanners, but it actually changes (dunno why)
3. ESPresence give the unique ID as ad:0xXXXX (where XXXX is part of the IRK)

Limitation:
The updates are not frequent, but I've not tested for long yet.

my 2 cents.

**Comment by franzbischoff** [comment]
Update, I've use the "query" on the watch, and it now gave me a "name:XXXXXXXX"

**Comment by kilian-f-za** [comment]
> Just to inform. It works!
> 
> 1. The IRK is broadcasted when you turn on the BT on the device
> 2. BT MAC seems fixed when you use Android App scanners, but it actually changes (dunno why)
> 3. ESPresence give the unique ID as ad:0xXXXX (where XXXX is part of the IRK)
> 
> Limitation:
> The updates are not frequent, but I've not tested for long yet.
> 
> my 2 cents.

Which app do you use? Does the app have to be installed on your phone or on the watch? I have a Galaxy Watch 4 and there it works, but my girlfriend's Galaxy Watch 4 doesn't work. Do you have a tip for me? I wanted to try the IRK now.

**Reply by franzbischoff** [reply]
Hi, I don't use an app on my watch. The app I referred to earlier is a phone app that scans all around.
The watch sends from time to time a BLE signal, but I don't know exactly what triggers it.

The most reliable is to use your phone as a beacon transmitter, at least for now :)

**Reply by kilian-f-za** [reply]
How did you determine the irk from the watch? With my phone I got it over the new UI devices but I can't connect to the watch to the Espresence.

Does it work on your phone without an additional app?

**Reply by franzbischoff** [reply]
I just saw the IRK the app received.

**Comment by mterry63** [comment]
For those using this with Home Assistant app, the current beta version enables iBeacon support on Wear OS and I can confirm it works well with ESPresense. It will generate an iBeacon which you can get the UUID with a beacon receiving app like nRF. It will show up as a bonded device with low RSSI.
